### PR TITLE
[BUFR Converter] Update ATMS/AMSUA YAMLs 

### LINKED ISSUE #1220: [BUFR Converter] Update YAMLs for ATMS and AMSU-A so the IODA files generated can work with UFO seamlessly
## Description

Update YAMLs for ATMS and AMSU-A so the IODA files generated can work with UFO seamlessly

##To-Do 
**Configuration**
- test date is 2021080100 
- Use antenna temperature (Ta)
- ATMS is FFT remapped

**Test using fv3jedi_hofx_nomodel.x with the IODA files converted from BUFR converter**  
- Convert BUFR to IODA format using bufr2ioda.x with YAML
- Transform FV3 background to GeoVaL
- Run UFO HofX
- Run UFO HofX with QC (without thinning) 

**Check List**

BUFR Converter YAML -- Updated YAML files are in [feature/iodaconv_gdasapp](https://github.com/JCSDA-internal/ioda-converters/tree/feature/iodaconv_gdasapp)
- [x] [ATMS YAML](https://github.com/JCSDA-internal/ioda-converters/blob/feature/iodaconv_gdasapp/test/testinput/bufr_ncep_atms_ta_remap.yaml)  
- [x] [AMSUA (all except n15) YAML](https://github.com/JCSDA-internal/ioda-converters/blob/feature/iodaconv_gdasapp/test/testinput/bufr_ncep_1bamua_ta.yaml)
        Note: AMSUA n15 BUFR contains brightness temperature 

Investigate diag output from UFO 
- [x] ATMS HofX
- [x] ATMS HofX + QC
- [x] AMSUA HofX
- [x] AMSUA HofX + QC

### PR DESCRIPTION
**Pull request by emilyhcliu**
## Description

This PR proposes the following changes:

- Update YAML files for ATMS and AMSU-A 
- Update test output accordingly
- Remove unnecessary input variable for getting sensor angle from footprint number

See details in Issue #1220 

### Issue(s) addressed

Link the issues to be closed with this PR
- fixes #1220 

## Dependencies

None

## Impact

Please list the other repositories (if any) that this PR will require changes in (example below)

- [ ] saber
- [ ] ufo
- [ ] ...

### CODE REVIEW

File: src/bufr/BufrParser/Exports/Variables/SensorScanAngleVariable.cpp [hunk 1]
```diff
@@ -25,13 +25,13 @@ namespace
     namespace ConfKeys
     {
         const char* FieldOfViewNumber = "fieldOfViewNumber";
-        const char* HeightOfStation = "heightOfStation";
+//      const char* HeightOfStation = "heightOfStation";
```
Comment (BenjaminRuston) [inline src/bufr/BufrParser/Exports/Variables/SensorScanAngleVariable.cpp:28]: @emilyhcliu any reason why we're not just removing this line?  seems like lower down you are removing `HMSL` from `test/testinput/bufr_ncep_atms.yaml` for instance

File: test/testinput/bufr_ncep_atms_ta_remap.yaml [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+                minute: "*/MINU"
+                second: "*/SECO"
+ 
+        splits:
+          satId:
+            category:
+              variable: satelliteIdentifier
+              map:
+                _224: npp 
+                _225: n20
```
Comment (BenjaminRuston) [inline test/testinput/bufr_ncep_atms_ta_remap.yaml:87]: can we go ahead and add n21 without penalty?

**Review by BenjaminRuston — COMMENTED** [review]
would like to have a different error when the `atms_beamwidth.txt` file missing

we can have execution stop or require an additional option in the yaml to force it to continue?

File: test/testinput/bufr_ncep_esamua.yaml [hunk 3]
```diff
@@ -70,16 +77,17 @@ observations:
                 _3: metop-b
                 _4: metop-a
                 _5: metop-c
-#               _206: noaa-15
-                _209: noaa-18
-                _223: noaa-19
+#               _206: n15
+                _209: n18
+                _223: n19
```
Comment (BenjaminRuston) [inline test/testinput/bufr_ncep_esamua.yaml:82]: either NOAA16 or NOAA17 lasted like a year correct?  
should this just list all the possible ones?

File: src/bufr/BufrParser/Exports/Variables/SensorScanAngleVariable.cpp (hunk 1 above)
Comment (BenjaminRuston) [reply src/bufr/BufrParser/Exports/Variables/SensorScanAngleVariable.cpp:28]:
```suggestion
```

File: src/bufr/BufrParser/Exports/Variables/SensorScanAngleVariable.cpp [hunk 4]
```diff
@@ -25,13 +25,13 @@ namespace
...
         const char* ScanStart = "scanStart";
         const char* ScanStep = "scanStep";
     }  // namespace ConfKeys
 
     const std::vector<std::string> FieldNames = {ConfKeys::FieldOfViewNumber,
-                                                 ConfKeys::HeightOfStation,
+//                                               ConfKeys::HeightOfStation,
```
Comment (BenjaminRuston) [inline src/bufr/BufrParser/Exports/Variables/SensorScanAngleVariable.cpp:34]:
```suggestion
```

**Comment by BenjaminRuston** [comment]
@emilyhcliu I'm actually getting 3 failures on the HPCMP Narwhal machine in a gnu environment:

```The following tests FAILED:
	 98 - test_iodaconv_bufr_ncep_atms (Failed)
	114 - test_iodaconv_mhs_conv (Failed)
	115 - test_iodaconv_amsua_conv (Failed)
```

looks like tolerance maybe bitwise reproduction not happening?

```
98: Test command: /bin/bash "/p/home/ruston/work/JCSDA/JEDI/ioda-bundle-gnu/build/bin/iodaconv_comp.sh" "netcdf" "/p/home/ruston/work/JCSDA/JEDI/ioda-bundle-gnu/build/bin/bufr2ioda.x testinput/bufr_ncep_atms.yaml" "gdas.t00z.atms_n20.tm00.nc" "0.0"
98: Environment variables: 
98:  OMP_NUM_THREADS=1
98: Test timeout computed to be: 10000000
98: PRE-MAIN-INFO BufrParser: Parsing file ./testinput/gdas.t00z.atms.tm00.bufr_d
98: PRE-MAIN-INFO Executing Queries
98: PRE-MAIN-INFO Building Bufr Data
98: PRE-MAIN-INFO Exporting Data
98: PRE-MAIN-INFO Finished [0.058s]
98: [CRAYBLAS_WARNING] Application linked against multiple cray-libsci libraries
98: DIFFER : VARIABLE : sensorViewAngle : POSITION : [3] : VALUES : -42.735 <> -42.735
98: DIFFER : VARIABLE : sensorViewAngle : POSITION : [11] : VALUES : -33.855 <> -33.855
98: DIFFER : VARIABLE : sensorViewAngle : POSITION : [23] : VALUES : -20.535 <> -20.535
98: DIFFER : VARIABLE : sensorViewAngle : POSITION : [24] : VALUES : -19.425 <> -19.425
98: DIFFER : VARIABLE : sensorViewAngle : POSITION : [28] : VALUES : -14.985 <> -14.985
98: DIFFER : VARIABLE : sensorViewAngle : POSITION : [29] : VALUES : -13.875 <> -13.875
98: DIFFER : VARIABLE : sensorViewAngle : POSITION : [30] : VALUES : -12.765 <> -12.765
98: DIFFER : VARIABLE : sensorViewAngle : POSITION : [31] : VALUES : -11.655 <> -11.655
98: DIFFER : VARIABLE : sensorViewAngle : POSITION : [34] : VALUES : -8.325 <> -8.325
98: DIFFER : VARIABLE : sensorViewAngle : POSITION : [35] : VALUES : -7.215 <> -7.215
98: DIFFER : VARIABLE : sensorViewAngle : POSITION : [36] : VALUES : -6.105 <> -6.105
98: DIFFER : VARIABLE : sensorViewAngle : POSITION : [37] : VALUES : -4.995 <> -4.995
98: DIFFER : VARIABLE : sensorViewAngle : POSITION : [38] : VALUES : -3.885 <> -3.885
```

**Comment by BenjaminRuston** [comment]
even when using the non-zero tolerance `IODA_CONV_COMP_TOL` still fails on Cray using their libsci

what do you think @emilyhcliu 

```
98: PRE-MAIN-INFO Finished [0.06s]
98: [CRAYBLAS_WARNING] Application linked against multiple cray-libsci libraries
98: DIFFER : VARIABLE : sensorViewAngle : POSITION : [41] : VALUES : -0.554998 <> -0.554996 : PERCENT : 0.000236272
98: DIFFER : VARIABLE : sensorViewAngle : POSITION : [42] : VALUES : 0.555002 <> 0.555 : PERCENT : 0.000343665
98: DIFFER : VARIABLE : sensorViewAngle : POSITION : [43] : VALUES : 1.665 <> 1.665 : PERCENT : 7.87568e-05
98: DIFFER : VARIABLE : sensorViewAngle : POSITION : [137] : VALUES : -0.554998 <> -0.554996 : PERCENT : 0.000236272
98: DIFFER : VARIABLE : sensorViewAngle : POSITION : [138] : VALUES : 0.555002 <> 0.555 : PERCENT : 0.000343665
98: DIFFER : VARIABLE : sensorViewAngle : POSITION : [139] : VALUES : 1.665 <> 1.665 : PERCENT : 7.87568e-05
98: DIFFER : VARIABLE : sensorViewAngle : POSITION : [237] : VALUES : -0.554998 <> -0.554996 : PERCENT : 0.000236272
98: DIFFER : VARIABLE : sensorViewAngle : POSITION : [238] : VALUES : 0.555002 <> 0.555 : PERCENT : 0.000343665
98: DIFFER : VARIABLE : sensorViewAngle : POSITION : [239] : VALUES : 1.665 <> 1.665 : PERCENT : 7.87568e-05
98: DIFFER : VARIABLE : sensorViewAngle : POSITION : [333] : VALUES : -0.554998 <> -0.554996 : PERCENT : 0.000236272
98: DIFFER : VARIABLE : sensorViewAngle : POSITION : [334] : VALUES : 0.555002 <> 0.555 : PERCENT : 0.000343665
98: DIFFER : VARIABLE : sensorViewAngle : POSITION : [335] : VALUES : 1.665 <> 1.665 : PERCENT : 7.87568e-05
98: DIFFER : VARIABLE : sensorViewAngle : POSITION : [439] : VALUES : -0.554998 <> -0.554996 : PERCENT : 0.000236272
98: DIFFER : VARIABLE : sensorViewAngle : POSITION : [440] : VALUES : 0.555002 <> 0.555 : PERCENT : 0.000343665
98: DIFFER : VARIABLE : sensorViewAngle : POSITION : [441] : VALUES : 1.665 <> 1.665 : PERCENT : 7.87568e-05
98: DIFFER : VARIABLE : sensorViewAngle : POSITION : [535] : VALUES : -0.554998 <> -0.554996 : PERCENT : 0.000236272
98: DIFFER : VARIABLE : sensorViewAngle : POSITION : [536] : VALUES : 0.555002 <> 0.555 : PERCENT : 0.000343665
98: DIFFER : VARIABLE : sensorViewAngle : POSITION : [537] : VALUES : 1.665 <> 1.665 : PERCENT : 7.87568e-05
98: DIFFER : VARIABLE : sensorViewAngle : POSITION : [615] : VALUES : -0.554998 <> -0.554996 : PERCENT : 0.000236272
98: DIFFER : VARIABLE : sensorViewAngle : POSITION : [616] : VALUES : 0.555002 <> 0.555 : PERCENT : 0.000343665
98: DIFFER : VARIABLE : sensorViewAngle : POSITION : [617] : VALUES : 1.665 <> 1.665 : PERCENT : 7.87568e-05
98: DIFFER : VARIABLE : sensorViewAngle : POSITION : [711] : VALUES : -0.554998 <> -0.554996 : PERCENT : 0.000236272
98: DIFFER : VARIABLE : sensorViewAngle : POSITION : [712] : VALUES : 0.555002 <> 0.555 : PERCENT : 0.000343665
98: DIFFER : VARIABLE : sensorViewAngle : POSITION : [713] : VALUES : 1.665 <> 1.665 : PERCENT : 7.87568e-05
98: DIFFER : VARIABLE : sensorViewAngle : POSITION : [797] : VALUES : -0.554998 <> -0.554996 : PERCENT : 0.000236272
```

**Comment by BenjaminRuston** [comment]
in a rare case it is fine on the M1 Mac ;0)

so just the Cray EX having issues

**Review by BenjaminRuston — APPROVED** [review]
this too happy to just move forward, though getting differences when using the `cray-libsci` these are not critical and in the noise of a system

**Comment by PatNichols** [comment]
> this too happy to just move forward, though getting differences when using the `cray-libsci` these are not critical and in the noise of a system

@BenjaminRuston Maybe we should use the cray compiler flags for ieee-fp. Ioda does this so we can borrow them. This might fix it.

**Review by PatNichols — APPROVED** [review]

**Comment by PatNichols** [comment]
@rmclaren I would feel better merging this if you could review it. Thanks!

**Review by rmclaren — APPROVED** [review]

**Pull request merged**